services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +45,43 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  if(event == NULL) return ERR_CODE_INVALID_ARG;
+  
+  error_code_t errCode;
+  if(xQueueSend(thermalMgrQueueHandle, event, 0) == errQUEUE_FULL) {
+    return ERR_CODE_QUEUE_FULL; 
+  }
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  error_code_t errCode;
+
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_INTERRUPT;
+  LOG_IF_ERROR_CODE(thermalMgrSendEvent(&event));
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
   while (1) {
-    
+    error_code_t errCode;
+    thermal_mgr_event_t event;
+    float temp;
```

Comment:
usually, we want to minimize the scope of the variables. temp is only used inside the line 72's if statement scope.